calico/network-policy/policy-rules/log-rules.mdx

```diff
@@ -0,0 +1,206 @@
+
+Example of eBPF logs:
+```bash
+curl-5288    [000] ..s2.  3055.982021: bpf_trace_printk: cali527b0801ecb-E: policy ALLOWED proto 6 src 172.16.193.131:56042 dest 69.147.80.12:80
+
+<...>-4950    [001] ..s2.  2960.748464: bpf_trace_printk: cali527b0801ecb-E: policy ALLOWED proto 1 src 172.16.193.131:0 dest 8.8.8.8:8
+
+<...>-5288    [000] ..s2.  3055.954466: bpf_trace_printk: cali527b0801ecb-E: policy ALLOWED proto 17 src 172.16.193.131:36325 dest 10.43.0.10:53
+```
+
+## Log format
+
+$[prodname] BPF log uses a custom format to output the information, here is the break down of this format:
+
+![curl](/img/calico/curl-bpf-log.svg)
+
+1. Process Name and ID, Be aware that the process name might not directly correspond to the responsible process for the traffic.
+2. CPU Number, Kernel flags, timestamp, bpf print function used to log this message. 
+3. Interface name
+4. Traffic type, in this example Egress. (E= Egress, X= XDP, I= Ingress)
+5. Policy verdict (ALLOWED, DENIED)
+6. IP protocol number, in this example 6 is for TCP. (1 ICMP, 17 UDP)
+7. Source IP and Source port
+8. Destination IP and Destination port 
+
+</TabItem>
+<TabItem label="iptables/ipvs" value="iptables">
+By default, $[prodname] prefixes these log entries with `calico-packet`, making it easier to filter them. The location and method for accessing these logs can vary depending on your Linux distribution.
+
+The following commands are some of the usual places that you can search for the policy log outputs:
+```bash
+journalctl
+/var/log/syslog
+/var/log/kern.log
+```
+
+Example of iptable logs:
+```bash
+2024-11-19T12:15:03.023805-08:00 c1-control kernel: calico-packet: IN=cali527b0801ecb OUT=eth0 MAC=ee:ee:ee:ee:ee:ee:1e:0a:36:33:f5:09:08:00 SRC=172.16.193.134 DST=69.147.80.15 LEN=60 TOS=0x00 PREC=0x00 TTL=63 ID=58406 DF PROTO=TCP SPT=45588 DPT=80 WINDOW=64860 RES=0x00 SYN URGP=0
+
+2024-11-19T12:15:28.648818-08:00 c1-control kernel: calico-packet: IN=cali527b0801ecb OUT=eth0 MAC=ee:ee:ee:ee:ee:ee:1e:0a:36:33:f5:09:08:00 SRC=172.16.193.134 DST=8.8.8.8 LEN=84 TOS=0x00 PREC=0x00 TTL=63 ID=48210 DF PROTO=ICMP TYPE=8 CODE=0 ID=1 SEQ=1
+
+2024-11-19T12:15:02.990083-08:00 c1-control kernel: calico-packet: IN=cali527b0801ecb OUT=calid3446e883f0 MAC=ee:ee:ee:ee:ee:ee:1e:0a:36:33:f5:09:08:00 SRC=172.16.193.134 DST=172.16.193.133 LEN=96 TOS=0x00 PREC=0x00 TTL=63 ID=19241 DF PROTO=UDP SPT=42896 DPT=53 LEN=76
+```
+</TabItem>
+</Tabs>
+
+
+## How to
+
+:::caution
+For a complete and efficient observability solution, consider exploring the use of [Flow Logs](/calico-enterprise/latest/visibility/visualize-traffic) available in Calico Enterprise and Calico Cloud.
+
+Using log rules is not be the most effective way to achieve comprehensive observability. While network observability can be achieved by utilizing log actions, this approach may introduce significant performance overhead.
+::: 
```

Comment:
Add the link to the "Additional resources" and keep this one too?

Reply:
Just additional resources.

Reply:
So the idea here is if someone reads "flowlogs," which is a Calico enterprise thing, they can immediately click on it and go to the correspondent page to learn more.
Removing the hyperlink from that sentence and moving it only to the additional resources would make what we are talking about less visible since there will be no indication of a link at the bottom of the page for that footnote when the user is reading the sentence and when our user gets there they might not even remember what flowlogs was ...

Reply:
We can add these sorts of things to additional resources, where they are available for folks to check out if they're interested. But without unnecessarily distracting them from their main job: understanding log actions and how to use them.

I see that I wasn't entirely clear in my earlier note. It's not just the link. This admonition needs to go away. Best to condense to 1-2 sentences and add to additional resources.